### ISSUE TITLE
Have IJulia honor set_default_plot_format

### ISSUE THREAD

**Issue by knuesel** [dcjones/Gadfly.jl#1529]
It would be nice if `set_default_plot_format(:png)` was working in IJulia. Not sure if this is a bug or feature request.

**Comment by Mattriks** [comment]
There's an [example on Julia Discourse](https://discourse.julialang.org/t/can-i-choose-display-software/29510/3) of one way to enable `set_default_plot_format`.  I haven't tested that out on IJulia.  Note IJulia creates it's own display, at the top of the dispaly stack:
```julia
using Gadfly
import Base.Multimedia as mltm

# display stack:
mltm.display

3-element Vector{AbstractDisplay}:
 TextDisplay(Base.TTY(Base.Libc.WindowsRawSocket(0x0000000000000278) open, 0 bytes waiting))
 Gadfly.GadflyDisplay()
 IJulia.InlineDisplay()
```

as does VScode.  Gadfly shouldn't override those displays by default e.g. see #1311.  You may have to `popdisplay()` to get the discourse example working in IJulia.

**Comment by knuesel** [comment]
Thanks for the link, I couldn't figure out how to make `set_default_plot_format` do anything.

I was expecting something like in Plots.jl, where you can do `plot(..., fmt=:png)` to show a specific plot as PNG, or `default(fmt=:png)` to change the format of all plots. For the user this "just works" in IJulia without fiddling with displays.

It seems that Plots.jl does it by overloading `IJulia.display_dict`, see [here](https://github.com/JuliaPlots/Plots.jl/blob/e1e0f8d1c93c4d66d1a5e635f55461f219101700/src/init.jl#L77) and [here](https://github.com/JuliaPlots/Plots.jl/blob/e1e0f8d1c93c4d66d1a5e635f55461f219101700/src/ijulia.jl#L39). Maybe more a hack than a proper solution...

**Comment by Mattriks** [comment]
I can't tell from your posts if you have found (for Gadfly):
```julia
p = plot(...)
draw(PNG(8inch, 8inch), p)
```

**Comment by knuesel** [comment]
Yes, in IJulia I'm currently using `plot(...) |> PNG()` which works fine but must be done for each plot. So it would be nice to have an easy way to change the default, and there's already `set_default_plot_format` which doesn't do much currently. But maybe the Plots.jl way is not the right way and the default should be set through IJulia rather than Gadfly...
